__tests__/app.test.js

```diff
@@ -72,3 +72,71 @@ describe("/api/article/:article_id", () => {
       });
   });
 });
+
+const voteUpdate = {
+  inc_votes: 500,
+};
+
+describe("PATCH /api/articles/:article_id", () => {
+  test("status 200: responds with the updated article", () => {
+    return request(app)
+      .patch("/api/articles/3")
+      .send(voteUpdate)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toEqual({
+          article_id: 3,
+          votes: 500,
+          ...body.article,
+        });
+      });
+  });
+  test("status 400: Returns a bad request message when given a endpoint of wrong type", () => {
+    return request(app)
+      .patch("/api/articles/banana")
+      .send(voteUpdate)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+  test.only("status 404: Returns a Route Not found message when given endpoint of correct type but is otherwise invalid", () => {
+    return request(app)
+      .patch("/api/articles/9999")
+      .send(voteUpdate)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Route Not Found");
+      });
+  });
+  test("status 422: returns an incorrect input message when sent invalid data key", () => {
```

Comment:
This could be a 400, but after some reading myself, I like it. Nice find :)

Reply:
yes, its contentious, theres is disagreement on this when i looked into it. But it seems applicable and i wanted to try a different error :).